go/libraries/doltcore/sqle/tables.go

```diff
@@ -1335,17 +1335,43 @@ func (t *AlterableDoltTable) RewriteInserter(
 		// table to rewrite it we also truncate all the indexes. Much easier to get right.
 		for _, index := range oldSch.Indexes().AllIndexes() {
 			var colNames []string
-			for _, colName := range index.ColumnNames() {
+			prefixLengths := index.PrefixLengths()
+			for i, colName := range index.ColumnNames() {
 				if strings.ToLower(oldColumn.Name) == strings.ToLower(colName) {
 					colNames = append(colNames, newColumn.Name)
+					if len(prefixLengths) > 0 {
```

Comment:
We only need to check if column being modified has `prefixLengths`. The other columns should already be valid since they passed all the checks in the analyzer, and aren't changing, so no need to look at those.